File: app/src/main/java/com/example/ipmsg20/customobjects/Message.java
package com.example.ipmsg20.customobjects;

import com.example.ipmsg20.room.Database;
import com.example.ipmsg20.room.MessageDao;

import java.net.InetAddress;

public class Message {

    public int first = 0;
    public int index = 0;
    public String sender = null;
    public String sender_group = null;
    public String msg = null;
    public InetAddress address = null;

    public Message() {}

    public Message parse(String msg_string) {

        String[] msg_parts = msg_string.split(":");
        this.first = Integer.parseInt(msg_parts[0]);
        this.index = Integer.parseInt(msg_parts[1]);
        this.sender = msg_parts[2];
        this.sender_group = msg_parts[3];
        this.msg = msg_parts[4];
        return this;
    }

    public void setAddress(InetAddress address) {
        this.address = address;
    }

    public void save(Database db) {
        MessageDao dao = db.messageDao();
        dao.addMessage( new com.example.ipmsg20.room.Message(
                        0,
                        String.valueOf(first),
                        sender,
                        sender_group,
                        address.toString()
                )
            );
    }

}

File: app/src/main/java/com/example/ipmsg20/room/Message.kt
package com.example.ipmsg20.room

import androidx.room.ColumnInfo
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Entity
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.PrimaryKey
import androidx.room.Query
import java.net.InetAddress

@Entity
data class Message(

    @PrimaryKey(autoGenerate = true) @ColumnInfo(name = "mid") val mid: Int,
    @ColumnInfo(name = "index") val index: String,
    @ColumnInfo(name = "sender") val sender: String,
    @ColumnInfo(name = "sender_group") val group: String,
    @ColumnInfo(name = "address") val address: String

)

@Dao
interface MessageDao {

    @Query("select * from Message")
    fun getAll(): List<Message>

    @Query("select * from Message where sender = :sender")
    fun getBySender(sender: String): List<Message>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun addMessage(message: Message): Long

    @Delete
    fun deleteMessage(message: Message)

}

class MessageRepo {
    suspend fun insertMessage(db: Database, message: Message) : Long {
        val dao = db.messageDao();
        return dao.addMessage(message)
    }
}